articles/implementing-firebase-in-flutter/index.md

```diff
@@ -0,0 +1,315 @@
+We'll create a stateful widget and scaffold, as shown below.
+
+```dart 
+class HomePage extends StatefulWidget {
+  @override
+  _HomePageState createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold();  // add scaffold here
+  }
+}
+```
+
+We can then create the above `UI` in this file by pasting the following code in the body section of the `Scaffold.` 
+
+```dart
+Scaffold(
+  appBar: AppBar(title: Text("Firebase Demo"),),
+  body: Container(
+  child: Column(
+    children: <Widget>[
+      SizedBox(height: 250.0),
+      Padding(
+        padding: EdgeInsets.all(10.0),
+        child: TextField(),
+      ),
+      SizedBox(height: 30.0),
+      Center(
+        child: RaisedButton(
+          color: Colors.pinkAccent,
+          child: Text("Save to Database"),
+          onPressed: () {
+            //call method flutter upload
+          }
+        )
+      ),
+    ],
+  ),
+),);
+```
+
+Import the `HomePage` class in the `main.dart` file. Ensure that the `HomePage` is initialized, as shown below.
+
+```dart
+import 'package:flutter/material.dart';
+import 'home.dart';
+
+void main(){
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  // This widget is the root of your application.
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Flutter Demo',
+      theme: ThemeData(
+        primarySwatch: Colors.blue,
+        visualDensity: VisualDensity.adaptivePlatformDensity,
+      ),
+      home: HomePage(),
+    );
+  }
+}
+```
+
+Ensure that you have the following imports at the Top of the Home page.
+```dart
+import 'package:flutter/material.dart';
+import 'package:firebase_database/firebase_database.dart';
+import 'package:firebase_core/firebase_core.dart';
+```
+
+- `package:flutter/material.dart` - This import allows you to access Flutter's built-in widgets and other functionalities.
+- `package:firebase_database/firebase_database.dart` - This import helps access the Firebase Realtime database features.
+- `package:firebase_core/firebase_core.dart` - This dependency enables you to connect to different Firebase products. 
+
+### Storing data
+We will initialize a Firebase `databaseReference` object and use it to store data.
+
+```dart
+final databaseRef = FirebaseDatabase._instance_.reference(); //database reference object
+
+void addData(String data) {
+  databaseRef.push().set({'name': data, 'comment': 'A good season'});
+}
+```
+
+The `addData` function stores the user input along with a string (`'comment': 'A good season'`). Essentially, we can add any object to the database.
+
+We'll call the `addData` function when the `RaisedButton` is clicked.
+
+```dart
+onPressed: () {
+ addData(textcontroller.text);
+  //call method flutter upload
+})),
+```
+We will access the user input using a `textcontroller`. This element allows us to listen for changes in a TextField.
```

Comment:
Please provide a code snippet on how to add a Text Controller state and where to add it.